### DISCUSSION TITLE
Adguard Home and Active Directory

### DISCUSSION THREAD

**Discussion by bert3ch** [AdguardTeam/AdGuardHome#7249] [Q&A] [answered]
Hey everyone,

I have a setup with an Active Directory (AD) server and another VM running AdGuard Home . As many of you know, for a PC to properly join the AD domain, its primary DNS should point to the DC IP.

The issue I’m facing is that even though I set AdGuard Home as the secondary DNS on the client machines, it doesn’t seem to be blocking ads or using the filters properly. It’s like the secondary DNS isn’t being utilized as expected.
I have added Adguard IP adress Forwarders on DC`s DNS settings but still same. I did research but  I could not figure it out.

Any advice on how to make both AD and AdGuard work harmoniously in this setup? 

Thanks a lot in advance. I’m new at this and could really use some guidance!

**Comment by Dynamic5912** [comment]
Drop AD and move to Azure ;)

**Comment by donald2612** [comment, marked as answer]
![image](https://github.com/user-attachments/assets/73267eb4-96e7-4561-addd-88864a39e165)

I would only put your Adguard-servers in the client DNS setup, so you would directly get an answer for internet-adresses. Then put conditional forwarders in the Adguard-setup for the domain like:
![image](https://github.com/user-attachments/assets/76814ddd-0d9b-4c65-a140-e9233a9b4886)

![image](https://github.com/user-attachments/assets/3467a568-2554-44ae-92c4-a2be5979fde5)

And disable IPv6 on your DCs or at least disable listening on IPv6-DNS.
![image](https://github.com/user-attachments/assets/ad86d63d-15bf-4b5f-8d5e-4f45d2d3f273)

That setup is working flawlessly since years here.

**Reply by bert3ch** [reply]
Hello @donald2612,

I have follow the guide and I think is It is working. Thank you very much.
But one thing, lets say if I put [/50.168.192.in-addr.arpa/]192.168.50.x. for private  reverse dns server, upstream test is passed but it doesn't allow me to apply. so I jsut put ip adress and it looks working with this way too.
![upstream](https://github.com/user-attachments/assets/7da74907-d163-4b09-a3ad-e35767e04104)
any suggestions?

**Reply by donald2612** [reply]
AdGuard wants ONE universal reverse-DNS without "condition" for reverse. So if you only have one subnet in which you AD-servers reside, it is enough to just write 192.168.50.1 so the direct IP-address of you AD-servers is there without any ([/50.168.192.in-addr.arpa/) subnet annotation.
Cheers!
